setup.py

```diff
@@ -496,7 +496,13 @@ def __init__(self, name, sse2=None, avx2=None, cpp11=None, cpp11_required=False,
             self.libraries.append('hdf5')
 
         else:
-            self.sources.append(os.path.join('src', 'hdf5_dl.c'))
+            if name.endswith("h5sz3") and sys.platform.startswith('darwin'):
+                # MacOS does not like to mix C and C++ code and next line
+                # does not work for the macro CALL
+                #self.sources.append(os.path.join('src', 'hdf5_dl.cpp'))
+                pass
+            else:
+                self.sources.append(os.path.join('src', 'hdf5_dl.c'))
```

Comment:
It should be possible to replace this with adding a static lib (see other comment) and that would remove the need for a specific handling for sz3 in the generic class.